### DISCUSSION TITLE
New Build and no response from stepper motors

### DISCUSSION THREAD

**Discussion by Ronsut** [grblHAL/core#293] [General]
Hi
I have just completed an open Build CNC lead screw router and tried the HALweb builder.
I used the generic 3 axis build with ganged Y axis. My board is one by Phill Barrett. See JSON

{"driver":"iMXRT1062","URL":"https://github.com/grblHAL/iMXRT1062","board":"BOARD_T41U5XBB","symbols":["USB_SERIAL_CDC=1","Y_GANGED=1","PROBE_ENABLE=1"],"docker_instance":"","grblHAL":"webBuilder"}

There is no response from any of the four stepper motors!

In the IO GUi settings under stepper (4). I made measurements of the Enable output voltage on the controller board. see below

  X,Y,Z boxes left unchecked, resulted in 5v on En after saving and restarting the controller board.
  X,Y,Z boxes checked resulted in on En 0v after saving and restarting the controller board. All Good.

However when I monitored the En voltage during jogging there was no change of the EN output voltage. I would have expected Ov on the En output of the T41U5XBB to have flipped to 5V but it stayed at 0V

Same result when i tried a gcode block (G0), relative move. In both cases tested 

One of the two tests, I thought should shown some response from the motor, what am I missing.

The BD6600 Modules all seem ok (green led on) 26V to each. En out on T41U5XBB to BD6600 driver Module EN+,  EN-  is returned to T41U5XBB GND
on each motor.

The limit switches trigger an alarm, the monitor window display is ok and shows status, verbose ok when checked, but there are no axis (XYZ) signals indicated in GUI.

**Comment by terjeio** [comment]
Settings please. `Help > About >To Clipboard` and paste in a comment.

> However when I monitored the En voltage during jogging there was no change of the EN output voltage. I would have expected Ov on the En output of the T41U5XBB to have flipped to 5V but it stayed at 0V

If _Step idle delay_ ($1) is set to 255 or _Steppers deenergize_ ($37) is XYZ (7) then the En outputs are not changing.

> The limit switches trigger an alarm, the monitor window display is ok and shows status, verbose ok when checked, but there are no axis (XYZ) signals indicated in GUI.

_Status report options_ ($10) controls which data is output in the real time report, is _Pin state_ checked? Default value is 511.

**Reply by Ronsut** [reply]
Thanks for prompt reply. I am in Australia, a different time zone.

The Pin state is set to 511, default state.

I have disconnected all but the Y and A axis drivers until I can get the ganged Y/A motors working. Is
that OK  

Ron


%
; grblHAL
; 1.1f.20230411
; [OPT:VNMSL2,35,1024,3,0]
; [NEWOPT:ENUMS,RT+,ES,REBOOT,TC,SED,RTC]
; [FIRMWARE:grblHAL]
; [NVS STORAGE:*FLASH]
; [DRIVER:iMXRT1062]
; [DRIVER VERSION:230331]
; [DRIVER OPTIONS:USB.1]
; [BOARD:T41U5XBB]
; [AUX IO:4,3,0,0]
;
$N0=
$N1=
; 0 - Step pulse time
$0=5.0
; 1 - Step idle delay
$1=25
; 2 - Step pulse invert
$2=0
; 3 - Step direction invert
$3=0
; 4 - Invert stepper enable pin(s)
$4=0
; 5 - Invert limit pins
$5=0
; 6 - Invert probe pin
$6=0
; 8 - Ganged axes direction invert
$8=0
; 10 - Status report options
$10=511
; 11 - Junction deviation
$11=0.010
; 12 - Arc tolerance
$12=0.002
; 13 - Report in inches
$13=0
; 14 - Invert control pins
$14=6
; 15 - Invert coolant pins
$15=0
; 16 - Invert spindle signals
$16=0
; 17 - Pullup disable control pins
$17=0
; 18 - Pullup disable limit pins
$18=0
; 19 - Pullup disable probe pin
$19=0
; 20 - Soft limits enable
$20=0
; 21 - Hard limits enable
$21=1
; 22 - Homing cycle
$22=0
; 23 - Homing direction invert
$23=0
; 24 - Homing locate feed rate
$24=25.0
; 25 - Homing search seek rate
$25=500.0
; 26 - Homing switch debounce delay
$26=250
; 27 - Homing switch pull-off distance
$27=1.000
; 28 - G73 Retract distance
$28=0.100
; 29 - Pulse delay
$29=0.0
; 30 - Maximum spindle speed
$30=1000.000
; 31 - Minimum spindle speed
$31=0.000
; 32 - Mode of operation
$32=0
; 33 - Spindle PWM frequency
$33=5000
; 34 - Spindle PWM off value
$34=0.0
; 35 - Spindle PWM min value
$35=0.0
; 36 - Spindle PWM max value
$36=100.0
; 37 - Steppers deenergize
$37=0
; 39 - Enable legacy RT commands
$39=1
; 40 - Limit jog commands
$40=0
; 43 - Homing passes
$43=1
; 44 - Axes homing, first pass
$44=4
; 45 - Axes homing, second pass
$45=3
; 46 - Axes homing, third pass
$46=0
; 62 - Sleep enable
$62=0
; 63 - Feed hold actions
$63=2
; 64 - Force init alarm
$64=0
; 65 - Probing feed override
$65=0
; 100 - X-axis travel resolution
$100=250.000
; 101 - Y-axis travel resolution
$101=250.000
; 102 - Z-axis travel resolution
$102=250.000
; 110 - X-axis maximum rate
$110=500.000
; 111 - Y-axis maximum rate
$111=500.000
; 112 - Z-axis maximum rate
$112=500.000
; 120 - X-axis acceleration
$120=10.000
; 121 - Y-axis acceleration
$121=10.000
; 122 - Z-axis acceleration
$122=10.000
; 130 - X-axis maximum travel
$130=200.000
; 131 - Y-axis maximum travel
$131=200.000
; 132 - Z-axis maximum travel
$132=200.000
; 341 - Tool change mode
$341=0
; 342 - Tool change probing distance
$342=30.0
; 343 - Tool change locate feed rate
$343=25.0
; 344 - Tool change search seek rate
$344=200.0
; 345 - Tool change probe pull-off rate
$345=200.0
; 346 - Restore position after M6
$346=1
; 370 - Invert I/O Port inputs
$370=0
; 372 - Invert I/O Port outputs
$372=0
; 384 - Disable G92 persistence
$384=0
; 398 - Planner buffer blocks
$398=35
; 481 - Autoreport interval
$481=0
%

**Reply by terjeio** [reply]
I've tested (with a scope) with your settings and all stepper outputs works as expected. _Signal_ "LEDs" works too - FYI these should not be lit in normal operation. It is odd that inverting the stepper enable signals changes the outputs but motion does not, could be a corrupted hex file? Zip it and post it for me to check.

**Comment by Ronsut** [comment]
Thanks for your support it is greatly appreciated.  
Hope the file helps. 

[CNCfirmware.zip](https://github.com/grblHAL/core/files/11295067/CNCfirmware.zip)

**Reply by terjeio** [reply]
Works for me. Can you measure the enable signal voltages directly on the Teensy pins?
And be aware that A and B axis motors (M3 and M4) has solder bridges that links them to X enable by default...

**Comment by Ronsut** [comment]
Will measure tomorrow and recheck the soldier bridges.  
Getting late here.

**Comment by Ronsut** [comment]
Results are as follows

Pin 10(STEN) on Teensy4.1 is 3.3 V in jog 0V in Idle (good)

En on Y axis is 4.31V in jog, 0V in idle (??? too low) 

En on Y axis but driver disconected shows 5.1V in jog mode 0V in idle. Tested with known good DB6600 drivers modules

There is no pulse out from from BOBoard (in Jog mode) checked with scope. (that concerns me)

I used the same settings that I sent you, now I know they work.

I will put together a new control box with the  existing T41U5XBB but make so I can easy swap it out for another T41U5XBB if need be. I will order Monday.

If I get the new box finished later this week I will run the steppers directly from the Teensy on each driver before I install the T41U5XBB. I want to be satified that the Teensy4.1  board is  Ok, (it was one I used for my SDR project). I have some level shifters around here somewhere.                                                 
I will update the thread.

**Comment by Ronsut** [comment]
I ordered the new board from P L Barret and received it yesterday. My CNC leadscrew router is working has I expected.  I have removed the two suspect driver chips from the first board and have ordered replacements SMD chips locally.  
Thanks for support. 
Ron
